### PR TITLE
Update tutorial-template-components-in-slot.asciidoc

### PR DESCRIPTION
**Pull request by gatanaso**
This PR updates the `tutorial-template-components-in-slot.asciidoc`. It applies the same changes to the doc as in #570 but on top of the latest master branch code.

---
This change is [Reviewable](https://reviewable.io/reviews/vaadin/flow-and-components-documentation/666)

### CODE REVIEW

File: documentation/polymer-templates/tutorial-template-components-in-slot.asciidoc [hunk 1]
```diff
@@ -48,28 +49,33 @@ public class ComponentContainer extends PolymerTemplate<TemplateModel> {
...
 ----
-
-Without the *slot* tag in the template the added label and button would not be visible to the user even though it can be located in the DOM.
-As we can see multiple components will show up in the slot when added to the template element that has an open `<slot></slot>`
+* Without the `<slot>` tag in the template, the label and button would not be visible to the user, even though they can be located in the DOM.
+* You can add multiple components and they will display in the slot when added to the template element that has an open `<slot></slot>`.
 
 [NOTE]
-Any element added to the light dom to be shown inside a `<slot>` can be removed from the dom and it will work as expected and disappear from the main element.
+You can remove any element from a `<slot>` and it will work as expected, and no longer be displayed in the main element.
```
Comment (mvysny) [inline documentation/polymer-templates/tutorial-template-components-in-slot.asciidoc]: Maybe reword to: 

You can remove any element from a `<slot>` - that will work as expected, causing the element no longer to be displayed in the main element.

File: documentation/polymer-templates/tutorial-template-components-in-slot.asciidoc [hunk 2]
```diff
@@ -48,28 +49,33 @@ public class ComponentContainer extends PolymerTemplate<TemplateModel> {
...
+
+== Using Default Content in the <slot> Element
 
-== Default content slots and named slots in templates
+The `<slot>` tag can contain default content that displays if nothing is set from the light DOM.
```
Comment (mvysny) [inline documentation/polymer-templates/tutorial-template-components-in-slot.asciidoc]: Suggestion:

The `<slot>` tag can contain default content that is displayed only if nothing else is set from the light DOM.

File: documentation/polymer-templates/tutorial-template-components-in-slot.asciidoc [hunk 3]
```diff
@@ -79,21 +85,24 @@ Slots can also be named so that only wanted content is added there. This is done
...
 [source,html]
 ----
 <slot name="fullName">
     <slot name="firstName"></slot>, <slot name="lastName"></slot>
 </slot>
 ----
 
-This will make the slot show data for light dom `slot="firstName"` `slot="lastName"` elements if no `slot="fullName"` is available and when a `<element slot="fullName">` is
-added it will "override/replace" the `firstName, lastName` data.
+* This makes the slot show data for light DOM `<slot="firstName">` and `<slot="lastName">` elements, if no `<slot="fullName">` element is available. If you add the `<slot="fullName">` element, it overrides and replaces the `firstName and lastName` data.
```
Comment (mvysny) [inline documentation/polymer-templates/tutorial-template-components-in-slot.asciidoc]: If you add element into the `<slot="fullName">` slot,

File: documentation/polymer-templates/tutorial-template-components-in-slot.asciidoc [hunk 4]
```diff
@@ -79,21 +85,24 @@ Slots can also be named so that only wanted content is added there. This is done
...
+
+The default slot and any named slot can contain multiple elements.
 
-The "default" slot and named slot can contain multiple elements.
+*Example*: Defining multiple elements for the default slot and multiple named slots.
```
Comment (mvysny) [inline documentation/polymer-templates/tutorial-template-components-in-slot.asciidoc]: I think that the original piece of info is quite important:

> The "default" slot and named slot can contain multiple elements.

This info is kinda lost in the new rewording.

File: documentation/polymer-templates/tutorial-template-components-in-slot.asciidoc [hunk 5]
```diff
@@ -48,28 +49,33 @@ public class ComponentContainer extends PolymerTemplate<TemplateModel> {
...
-For example the template could be built as:
 [source,html]
 ----
 <div style="border: 1px solid black; padding: 10px; margin: 10px;">
     <slot>No components added</slot>
 </div>
 ----
 
-This would show at start 'No components added' in the slot which would then be replaced by the first added child element.
+* 'No components added' displays initially, and is then replaced by the first added child element.
```
Comment (mvysny) [inline documentation/polymer-templates/tutorial-template-components-in-slot.asciidoc]: > replaced by the first added child element.

Only the first child element, or all child elements? Or perhaps do you mean that any added child element replaces the 'No components added'?

File: documentation/polymer-templates/tutorial-template-components-in-slot.asciidoc [hunk 6]
```diff
@@ -141,8 +151,8 @@ class MainLayout extends PolymerElement {
 customElements.define(MainLayout.is, MainLayout);
 ----
 
-To use the template file, you also need a basic template component class with a `@JsModule` annotation for the template file.
-Right now you need to also implement the `RouterLayout` interface:
+
+*Example*: Basic `MainLayout` server class that imports the Polymer template and implements `RouterLayout`.
```
Comment (mvysny) [inline documentation/polymer-templates/tutorial-template-components-in-slot.asciidoc]: The original remark importance is a bit lost by moving it into an 'Example' block. Many quick-readers skip anything that's after the 'Example' block. I'd keep the original, perhaps with a bit of rewording.

File: documentation/polymer-templates/tutorial-template-components-in-slot.asciidoc [hunk 7]
```diff
@@ -154,12 +164,11 @@ public class MainLayout extends PolymerTemplate<TemplateModel>
 ----
 
 [NOTE]
-The method `showRouterLayoutContent(HasElement)` in the `RouterLayout` interface 
-has default implementation so you don't need to write anything in addition to that.
-But you may reimplement it as you wish.
+The `showRouterLayoutContent(HasElement)` method in the `RouterLayout` interface has default implementations. This makes it unnecessary to write additional code, but you can re-implement it, if necessary.
```
Comment (mvysny) [inline documentation/polymer-templates/tutorial-template-components-in-slot.asciidoc]: implementation <- singular, only one implementation.

> but you can re-implement it, if necessary.

Perhaps: but you can override and re-implement it, if necessary.

File: documentation/polymer-templates/tutorial-template-components-in-slot.asciidoc (hunk 1 above)
Comment (gatanaso) [reply documentation/polymer-templates/tutorial-template-components-in-slot.asciidoc]: What do you think about something along these lines:
```
You can remove any element from a `<slot>` and it will work as expected, the element will no longer be displayed as part of the main element.
```

File: documentation/polymer-templates/tutorial-template-components-in-slot.asciidoc (hunk 2 above)
Comment (gatanaso) [reply documentation/polymer-templates/tutorial-template-components-in-slot.asciidoc]: I agree, will change the sentence!

File: documentation/polymer-templates/tutorial-template-components-in-slot.asciidoc (hunk 5 above)
Comment (gatanaso) [reply documentation/polymer-templates/tutorial-template-components-in-slot.asciidoc]: I believe the original author meant that the "no components added" text will be replaced as soon as there is at least one child element added, hence the "first". Though it is worded in a way that may cause others to get confused. Will modify it, thanks for pointing it out.

File: documentation/polymer-templates/tutorial-template-components-in-slot.asciidoc (hunk 3 above)
Comment (gatanaso) [reply documentation/polymer-templates/tutorial-template-components-in-slot.asciidoc]: Correct, will update. 👍

File: documentation/polymer-templates/tutorial-template-components-in-slot.asciidoc (hunk 6 above)
Comment (gatanaso) [reply documentation/polymer-templates/tutorial-template-components-in-slot.asciidoc]: Yes I agree. Will introduce back the remark.

File: documentation/polymer-templates/tutorial-template-components-in-slot.asciidoc (hunk 7 above)
Comment (gatanaso) [reply documentation/polymer-templates/tutorial-template-components-in-slot.asciidoc]: Agreed, will update!

File: documentation/polymer-templates/tutorial-template-components-in-slot.asciidoc (hunk 1 above)
Comment (mvysny) [reply documentation/polymer-templates/tutorial-template-components-in-slot.asciidoc]: Sounds good!

File: documentation/polymer-templates/tutorial-template-components-in-slot.asciidoc (hunk 4 above)
Comment (gatanaso) [reply documentation/polymer-templates/tutorial-template-components-in-slot.asciidoc]: Yeah, somewhat. Though, I would suggest we go with the original author's rewording on this one, if it's okay with you?
